### DISCUSSION TITLE
Implement `::` synthetic sugar for CAST

### DISCUSSION THREAD

**Discussion by buremba** [trinodb/trino#13768] [Ideas]
Postgresql and Snowflake have a nice synthetic sugar `val::type` that is similar to `CAST(val as type)` but a more human-readable version of it. In addition to that, Snowflake has `jsoncol:path` which extracts the `path` from JSON ([VARIANT](https://docs.snowflake.com/en/sql-reference/data-types-semistructured.html) type)

I believe it would be useful to support for Trino to support the `::` syntax sugar syntax.

**Comment by hashhar** [comment]
There's not very strong motivation to add this IMO.

It's not part of the ANSI SQL spec.
Notably very few databases use `::` for CAST.
Some databases use `::` for entirely different purpose (function namespaces and resolution).

Is there some reason other than familiarity that would be useful (e.g. does it make generation of SQL statements easier etc.)?

**Comment by buremba** [comment]
That's fair. Another reason why it might be useful is that even though it's not part of ANSI SQL spec, Snowflake & PG makes use of this in their docs so the people who use them are already familiar with it. It will make it easier for those people to get used to Trino.

**Comment by krutoileshii** [comment]
i'd love this to be an option, mostly because its shorted than cast syntax.

**Comment by raunaqmorarka** [comment]
This is being worked on in https://github.com/trinodb/trino/pull/25259
